### PR TITLE
synth: Introduce `-extra-map` for amending techmap

### LINKED ISSUE #4220: Have a way to inject extra techmapping rules when calling `synth`
### Feature Description

A great power of customization would be exposed if we had a way to inject user-written "techmap" rules into the internal `synth` script. This should take the form of a command option to `synth`.

### PR DESCRIPTION
**Pull request by povik**
Closes #4220

### CODE REVIEW

**Comment by povik** [comment]
Quoting the help message for the new option:

```
    -inject filename
        inject rules from the given file to complement the default
        mapping library in the `techmap` step. this option can be
        repeated.
```

**Comment by povik** [comment]
TODO: rename to `-extra-map`

File: techlibs/common/synth.cc [hunk 1]
```diff
@@ -261,7 +270,17 @@ struct SynthPass : public ScriptPass {
 			run("opt -fast -full");
 			run("memory_map");
 			run("opt -full");
-			run("techmap");
+			if (help_mode) {
+				run("techmap", "                  (unless -inject)");	
+				run("techmap -map +/techmap.v -map <inject>", "  (if -inject)");
```
Comment (nakengelhardt) [inline techlibs/common/synth.cc:275]:
```suggestion
				run("techmap", "                  (unless -extra-map)");	
				run("techmap -map +/techmap.v -map <inject>", "  (if -extra-map)");
```

Comment (povik) [reply techlibs/common/synth.cc]: Thanks!

**Pull request merged**